drivers/led/pca_common.c

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2018 Savoir-Faire Linux.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief LED driver for the PCA963x I2C LED drivers
+ */
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/zephyr.h>
+
+#define LOG_LEVEL CONFIG_LED_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pca_common);
+
+#include "led_context.h"
+
+#define _DT_PROP_ANY_OR(id, compat, prop) DT_PROP(DT_INST(id, compat), prop) ||
+#define DT_PROP_ANY(compat, prop) \
+	DT_FOREACH_STATUS_OKAY_VARGS(compat, _DT_PROP_ANY_OR, compat, prop) 0
+
+#define _DT_ANY_NODE_HAS_PROP_OR(id, compat, prop) DT_NODE_HAS_PROP(DT_INST(id, compat), prop) ||
+#define DT_ANY_NODE_HAS_PROP(compat, prop) \
+	DT_FOREACH_STATUS_OKAY_VARGS(compat, _DT_ANY_NODE_HAS_PROP_OR, compat, prop) 0
```

Comment:
Well... it is quite something :) I suppose because `external-driver` can be false, then we can't use `DT_ANY_NODE_HAS_PROP` with it.... How many bytes of memory are we saving with this trick ? I'm not sure it's worth the readability we give. But the code looks correct.